src/GraphQl/Type/TypeBuilderEnumInterface.php

```diff
@@ -31,9 +30,9 @@ interface TypeBuilderEnumInterface
     /**
      * Gets the object type of the given resource.
      *
-     * @return ObjectType|NonNull the object type, possibly wrapped by NonNull
+     * @return GraphQLType the object type, possibly wrapped by NonNull
      */
-    public function getResourceObjectType(?string $resourceClass, ResourceMetadataCollection $resourceMetadataCollection, Operation $operation, bool $input, bool $wrapped = false, int $depth = 0): GraphQLType;
+    public function getResourceObjectType(?string $resourceClass, ResourceMetadataCollection $resourceMetadataCollection, Operation $operation, bool $input, bool $wrapped = false, int $depth = 0, ApiProperty $propertyMetadata = null): GraphQLType;
```

Comment:
Because of the BC promise (https://symfony.com/doc/current/contributing/code/bc.html) we can not add an argument here. I think that the best here would be to create a new interface. 

Example: 

```php
interface ContextAwareTypeBuilderInterface
{
    /**
     * Gets the object type of the given resource.
     *
     * @param array<string, mixed>&array{input?: bool, wraped?: bool, depth?: int} $context
     *
     * @return GraphQLType the object type, possibly wrapped by NonNull
     */
    public function getResourceObjectType(ResourceMetadataCollection $resourceMetadataCollection, Operation $operation, ?ApiProperty $apiProperty = null, array $context = []): GraphQLType;

    /**
     * Get the interface type of a node.
     */
    public function getNodeInterface(): InterfaceType;

    /**
     * Gets the type of a paginated collection of the given resource type.
     */
    public function getPaginatedCollectionType(GraphQLType $resourceType, Operation $operation): GraphQLType;

    /**
     * Gets the type corresponding to an enum.
     */
    public function getEnumType(Operation $operation): GraphQLType;

    /**
     * Returns true if a type is a collection.
     */
    public function isCollection(Type $type): bool;
}

```

I changed the method to be more future proof, we can always use the `context` to avoid changing (again) this interface... `$resourceClass` is always available through ($operation->getClass()) and `input`, `wrapped` and `depth` are inside the context. 

Then you need to deprecate the `TypeBuilderEnumInterface` and to check inside the `TypeConverter` if the instance is a `ContextAwareTypeBuilderInterface` so we can call the new signature. Sorry I didn't catched that earlier on